### ISSUE TITLE
Make an RFC for embedding types, contracts, and other specifications

### ISSUE THREAD

**Issue by jeapostrophe** [racket/rhombus#9]
A key idea, for me, is to be able to separate specifications from the code that uses them, so multiple modules can use the same specification. Here specification is types, contracts, tests, theorems that are true about the software, and so on.

**Comment by samdphillips** [comment]
Can you see this as a mechanism to compile code without contracts?

**Comment by jeapostrophe** [comment]
Yup. My perspective is that we should write down very expressive specifications (theorems and contracts with lots of dependencies and precision) and be able to smoothly compile them in a variety of policies from full enforcement and testing to no-seatbelts-or-helmets.

**Comment by mflatt** [comment]
Categorizing as "yep moving on" on the grounds that annotations fit this idea.
